### PR TITLE
GT-1589 shareable analytics

### PR DESCRIPTION
**Pull request by rachaelblue**
https://jira.cru.org/browse/GT-1589

### CODE REVIEW

File: godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/ShareableImageModel.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  ShareableImageModel.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 7/25/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import UIKit
+
+struct ShareableImageModel {
```
Comment (levieggertcru) [inline godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/ShareableImageModel.swift:11]: Hey @rachaelblue can you change this name to ```ShareableImageDomainModel```.  That is something I would like to shift towards for our domain and data models.  Suffixing with either ```DomainModel``` or ```DataModel```.  That will help us see which is coming from the domain vs data layer.

File: godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    
+    func getShareableImage(from shareable: Shareable, manifestResourcesCache: ManifestResourcesCache) -> ShareableImageModel? {
+        guard let shareableImage = shareable as? ShareableImage, let resource = shareableImage.resource, let imageToShare = manifestResourcesCache.getImageFromManifestResources(resource: resource) else {
+            return nil
+        }
+        
+        return ShareableImageModel(
+            image: imageToShare,
+            imageId: shareable.id,
+            toolAbbreviation: shareable.manifest.code
```
Comment (levieggertcru) [inline godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift:22]: I noticed that the ```manifest.code``` is an optional value.  Would it be safer to fetch the abbreviation from the resource?  Or fallback to the resource abbreviation if manifest code is null?  @frett thoughts here.

**Review by levieggertcru — CHANGES_REQUESTED** [review]

File: godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift (hunk 2 above)
Comment (frett) [reply godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift:22]: on Android I just pass it as null if it doesn't exist. We've been actively publishing the code and the locale in manifests for several years now, it's just older translations that may be missing them.

If you want to fallback to the what is set in your local resource, that's fine

Comment (frett) [reply godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift:22]: https://github.com/CruGlobal/mobile-content-api/pull/159

For reference, we've been including those attributes for almost 4 years now when publishing a translation

Comment (frett) [reply godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift:22]: https://github.com/CruGlobal/mobile-content-api/pull/159

For reference, we've been including those attributes for almost 4 years now when publishing a translation

Comment (levieggertcru) [reply godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift:22]: I'm fine with mimicking Android here.

File: godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/ShareableImageModel.swift (hunk 1 above)
Comment (rachaelblue) [reply godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/ShareableImageModel.swift:11]: Ooo good idea, I was thinking about that too.  That'll help make things clearer for sure.

**Review by levieggertcru — APPROVED** [review]

**Pull request merged**